ui/pages/confirmations/components/alerts/alert-modal/alert-modal.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useCallback } from 'react';
+import { ButtonVariant } from '@metamask/snaps-sdk';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  Checkbox,
+  Icon,
+  IconName,
+  IconSize,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '../../../../../components/component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BlockSize,
+  BorderRadius,
+  Display,
+  IconColor,
+  Severity,
+  TextAlign,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import useAlerts from '../../../hooks/useAlerts';
+
+export type AlertModalProps = {
+  /** The unique identifier of the entity that owns the alert */
+  ownerId: string;
+  /** The function to be executed when the button in the alert modal is clicked */
+  handleButtonClick: () => void;
```

Comment:
Should this be `on` rather than `handle`?

Plus as we'll have dynamic buttons in a future ticket, should we be explicit and call this `onAcknowledgeClick`?

It may be that the `Got it` button itself becomes dynamic but for now would it add clarity?